app/src/main/java/com/keylesspalace/tusky/util/LinkHelper.kt

```diff
@@ -322,3 +317,46 @@ fun looksLikeMastodonUrl(urlString: String): Boolean {
 }
 
 private const val TAG = "LinkHelper"
+
+/**
+ * [LinkMovementMethod] that doesn't add a leading/trailing clickable area.
+ *
+ * [LinkMovementMethod] has a bug in its calculation of the clickable width of a span on a line. If
+ * the span is the last thing on the line the clickable area extends to the end of the view. So the
+ * user can tap what appears to be whitespace and open a link.
+ *
+ * Fix this by overriding ACTION_UP touch events and calculating the true start and end of the
+ * content on the line that was tapped. Then ignore clicks that are outside this area.
+ *
+ * See https://github.com/tuskyapp/Tusky/issues/1567.
+ */
+class NoTrailingSpaceLinkMovementMethod : LinkMovementMethod() {
+    override fun onTouchEvent(widget: TextView?, buffer: Spannable?, event: MotionEvent?): Boolean {
+        widget ?: return super.onTouchEvent(widget, buffer, event)
+        val action = event?.action ?: return super.onTouchEvent(widget, buffer, event)
+        if (action != ACTION_UP) return super.onTouchEvent(widget, buffer, event)
+
+        val x = event.x.toInt()
+        val y = event.y.toInt() - widget.totalPaddingTop + widget.scrollY
+        val line = widget.layout.getLineForVertical(y)
+        val lineLeft = widget.layout.getLineLeft(line)
+        val lineRight = widget.layout.getLineRight(line)
+        if (x > lineRight || x >= 0 && x < lineLeft) {
+            return true
+        }
+
+        return super.onTouchEvent(widget, buffer, event)
+    }
+
+    companion object {
+        private var instance: NoTrailingSpaceLinkMovementMethod? = null
+
+        fun getInstance(): NoTrailingSpaceLinkMovementMethod {
+            if (instance == null) {
+                instance = NoTrailingSpaceLinkMovementMethod()
+            }
+
+            return instance!!
+        }
+    }
```

Comment:
make `NoTrailingSpaceLinkMovementMethod` a Kotlin `object` then this whole business is not needed

Reply:
Good idea. I've kept the `getInstance()` method for consistency with `LinkMovementMethod`.